packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -304,10 +312,38 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
     if (!_enabled) {
       return false;
     }
+
     final loggingConstraint = _getLoggingConstraint();
+    if (loggingConstraint.userLogLevel.containsKey(_userId)) {
+      final userLogLevel = loggingConstraint.userLogLevel[_userId]!;
+      final userCategoryLogLevel =
+          _getCategoryLogLevel(userLogLevel.categoryLogLevel, logEntry);
+      if (userCategoryLogLevel != null) {
+        return logEntry.level >= userCategoryLogLevel;
+      }
+      return logEntry.level >= userLogLevel.defaultLogLevel;
+    }
+
+    final categoryLogLevel =
+        _getCategoryLogLevel(loggingConstraint.categoryLogLevel, logEntry);
+    if (categoryLogLevel != null) {
+      return logEntry.level >= categoryLogLevel;
+    }
     return logEntry.level >= loggingConstraint.defaultLogLevel;
   }
```

Comment:
```suggestion
    final loggingConstraint = _getLoggingConstraint();

    final logLevel = loggingConstraint.userLogLevel.containsKey(_userId)
        ? _getCategoryLogLevel(
            loggingConstraint.userLogLevel[_userId]!.categoryLogLevel,
            logEntry,
          )
        : _getCategoryLogLevel(loggingConstraint.categoryLogLevel, logEntry);

    if (logLevel != null) {
      return logEntry.level >= logLevel;
    }

    return logEntry.level >= loggingConstraint.defaultLogLevel;
```
suggestion: try flattening nested if statements and reusing the comparison logic. This is more readable imo